GTFS/Entities/TimeOfDay.cs

```diff
@@ -71,6 +71,58 @@ public static TimeOfDay FromTotalSeconds(int totalSeconds)
             };
         }
 
+        public static TimeOfDay FromDateTime(DateTime date)
+        {
+            return FromTimeSpan(date.TimeOfDay);
+        }
+
+        public static TimeOfDay FromTimeSpan(TimeSpan ts)
+        {
+            return FromTotalSeconds(Convert.ToInt32(ts.TotalSeconds));
+        }
+
+        /// <summary>
+        /// Creates a new time of day from a string in the HH:MM:SS format.
+        /// </summary>
+        public static TimeOfDay FromString(string timeString)
+        {
+            try
+            {
+                TimeOfDay tod = new TimeOfDay();
+                var tokens = timeString.Split(':');
+                for (int i = 0; i < tokens.Length; i++)
+                {
+                    if (tokens[i].Contains(" "))
+                    {
+                        tokens[i] = tokens[i].ToCharArray()[0] + "0";
+                    }
+                    else if (tokens[i] == "" || tokens[i] == "  ")
```

Comment:
Second condition of this else if should never happen right? Can't remember writing this but apparently it was me, guess if it works right all good. Maybe remove the stuff after the || since I can't see how it would ever happen

Reply:
I think I wrote this. All the extra looking at spaces was just to add zeros where the user didn't add them in the masked textbox

Reply:
going to redo this for loop

Reply:
seems this is the way it has to be. seems masked textboxes can be tricky with empty tokens.

Reply:
Yeah, it's weird though because if tokens[i] == "  " then it must be true that tokens[i].Contains(" ") which means the first condition will be executed which is why the bit after the || doesn't make sense to me. Unless I'm going crazy

Reply:
ah I see. will fix it now. it's a wonder it worked till now